obspy/io/sac/util.py

```diff
@@ -159,19 +159,24 @@ def is_same_byteorder(bo1, bo2):
     return (bo1.lower() in le) == (bo2.lower() in le)
 
 
-def _clean_str(value):
+# This fn is used in two places: in SACTrace.read, to sanitize strings for
+# SACTrace, and in sac_to_obspy_header, to sanitize strings for Trace that the
+# user may have manually added.
+def _clean_str(value, strip_whitespace=True):
     # Remove null values and whitespace, return a str
     try:
         # value is a str
         null_term = value.find('\x00')
     except TypeError:
         # value is a bytes
-        # null_term = value.decode().find('\x00')
         null_term = value.find(b'\x00')
+    except UnicodeError:
+        null_term = value.decode('ASCII', 'replace').find(b'\x00')
 
     if null_term >= 0:
-        value = value[:null_term]
-    value = value.strip()
+        value = value[:null_term] + b" " * len(value[null_term:])
```

Comment:
![screenshot from 2016-07-28 15 35 47](https://cloud.githubusercontent.com/assets/1842780/17214407/144781da-54d9-11e6-87b1-0b33b7a0de6f.png)

n.b. only failing on Python3, Python2 is doing auto de/encoding when possible.